app/Community/Components/UserProfileMeta.php

```diff
@@ -425,19 +426,30 @@ private function calculateAverageFinishedGames(array $userJoinedGamesAndAwards):
         return number_format($averageFinishedGames, 2, '.', '');
     }
 
-    private function calculateAveragePointsPerWeek(string $userMemberSince, int $points = 0): int
+    private function calculateAveragePointsPerWeek(User $user, bool $hardcore, int $points = 0): int
     {
-        $memberSince = Carbon::createFromFormat('Y-m-d H:i:s', $userMemberSince);
+        $field = $hardcore ? "unlocked_hardcore_at" : "unlocked_at";
+
+        $startingDate = $user
+            ->playerAchievements()
+            ->whereNotNull($field)
+            ->orderBy($field)
+            ->first()[$field] ?? null;
```

Comment:
This could be further optimized. `[$field]` kind of clues in that Eloquent ORM is doing `SELECT *` under the hood. What we really want is `SELECT $field`.

```php
$startingDate = $user->playerAchievements()
    ->whereNotNull($field)
    ->orderBy($field, 'asc')
    ->value($field);
```

A minor nit. This is a strange convention coming from another language. Typically in this codebase, FP chains are formatted like:

```php
$result = $subject->initialFunction()
    ->functionB()
    ->functionC()
    ->functionD();
```

It took me a while to get used to this. Unfortunately, our automated tooling doesn't fix this formatting.

Reply:
Oh, I should really have read the Eloquent documentation more carefully. Oh, well, I learnt something today.